spec/support/expectations/equal_namespace_definitions.rb

```diff
@@ -0,0 +1,55 @@
+# frozen_string_literal: true
+
+require 'rspec/expectations'
+
+RSpec::Matchers.define :have_equal_namespace_definitions_as do |expected|
+  attr_reader :actual, :expected
+
+  match do |actual|
+    return false unless actual.is_a?(String) && expected.is_a?(String)
+
+    expected_xml = parse_fragment(expected)
+    @expected = expected_xml.to_xml
+    actual_xml = parse_fragment(actual)
+    @actual = actual_xml.to_xml
+
+    return false if expected_xml.errors.length.positive? || actual_xml.errors.length.positive?
+    return false unless EquivalentXml.equivalent?(expected_xml, actual_xml)
+
+    return compare_namespaces(actual_xml.children.first, expected_xml.children.first)
+  end
+
+  failure_message do |actual|
+    "#{@actual || actual} is not equal to \n#{@expected || expected}."
```

Comment:
I think it would be nice to have a more accurate error message (ie "does not have equal namespace as")